src/python/WMCore/WorkQueue/WMBSHelper.py

```diff
@@ -348,18 +349,14 @@ def addMCFakeFile(self):
             if self.mask and self.mask.get(key) is None:
                 msg = 'Invalid value "%s" for %s' % (self.mask.get(key), key)
                 raise WorkQueueWMBSException(msg)
+
         locations = set()
-        for site in self.getLocations.execute(conn=self.getDBConn(),
-                                              transaction=self.existingTransaction()):
-            try:
-                siteInfo = self.getLocationInfo.execute(site, conn=self.getDBConn(),
-                                                        transaction=self.existingTransaction())
-                if not siteInfo:
-                    self.logger.info('Skipping MonteCarlo injection to site "%s" as unknown to wmbs' % site)
-                    continue
-                locations.add(siteInfo[0]['pnn'])
-            except Exception as ex:
-                self.logger.error('Error getting storage element for "%s": %s' % (site, str(ex)))
+        siteInfo = self.getLocationInfo.execute(conn=self.getDBConn(),
+                                                transaction=self.existingTransaction())
+        for site in siteInfo:
+            if site['pnn'] in self.commonLocation:
```

Comment:
What happens if sefl.commonLocation is `[]` [1]. Do we skip submission? Or we are assuring that for those workflows we always provide this parameter to the call? Do we have such a check outside the WMBSHelper class (where it is actually instanced)? The non `None` value here seems strange to me to be the only way of communicating the fact that we are running on the particular type of workflows which need the additional check here.
   
https://github.com/dmwm/WMCore/pull/9659/commits/964437b2a607f63074e3f46ded6e2758b4cb65f7#diff-af5cb316c0b66783a9f738da4bc8c83aR173

Reply:
If `self.commonLocation` is empty, then work will failed to get acquired in WMBS.
However, there shouldn't be such cases, otherwise it would mean that WQEs would never leave the global workqueue.

Reply:
My point exactly. My question was regarding this part - 'there shouldn't be such cases'. Is there an explicit check for that outside this class? I am sure you already anticipated that kind of situation, but thought would be faster to ask instead of digging into the code myself.

Reply:
This piece of code is actually getting work from the local queue (LQE) and adding it to the WMBS. That means, if the LQE cannot fulfill all the data locations, work cannot be materialized in jobs (WMBS insertion). In other words, there is no explicit check for self.commonLocation, but I do thing the check on common locations for WQEs cover it.

Reply:
That's fine with me. It doesn't matter where the check is done, as long as you are sure we do not create like a memory leak behavior in terms of LQEs ... I mean to start accumulating elements in the LQ which never get submitted for real. All the rest is good I believe. Resolving this conversation now so we can continue with more important stuff. Thank you @amaltaro  !